### PR TITLE
lib: homeManagerConfiguration support for direct home-manager support

### PR DESCRIPTION
**Pull request by kalbasit**
Usage:

```
$ home-manager {build,switch} --flake <attribute>
```

Instructions testing this in a VM (recommended):

1. Clone my soxincfg at `add-home-manager-support` ([PR](https://github.com/kalbasit/soxincfg/pull/14)).
   ```
   git clone -b add-home-manager-support git@github.com:kalbasit/soxincfg.git
   ```
2. Switch to my Crostini configuration (you can try it in the VM)
    ```
    home-manager switch --flake .#penguin
    ```

NOTE: This was extracted from #29 and tested only in that branch.
NOTE: This PR depends on #36

### CODE REVIEW

**Comment by kalbasit** [comment]
I have not added the example configuration because I'm still working on it. I intend to leave this task for later as I'm planning to add support for sops-nix to Soxin and then copy my config  (skeleton of it) as the template.

File: lib/home-manager-configuration.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{ self, lib, home-manager }:
+
+{ configuration
+, system
+, homeDirectory
+, username
+, modules ? [ ]
```
Comment (rissson) [inline lib/home-manager-configuration.nix:7]: this doesn't match what's done in #32. Let's be consistent in the way we name things, even if that means some extra code on our side.

File: lib/home-manager-configuration.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{ self, lib, home-manager }:
+
+{ configuration
+, system
+, homeDirectory
+, username
```
Comment (rissson) [inline lib/home-manager-configuration.nix:6]: Some of those aren't used below.

**Review by rissson — CHANGES_REQUESTED** [review]

File: lib/home-manager-configuration.nix (hunk 1 above)
Comment (kalbasit) [reply lib/home-manager-configuration.nix:7]: You mean global and hm? hmm do you think we will ever have two+ systems in home-manager to merit a split?

Comment (rissson) [reply lib/home-manager-configuration.nix:7]: We will probably never have something else than home-manager in home-manager configurations, but what if I want to do something like this in my soxincfg:

```
hmModules = [ /* ... */ ];
nixosHosts = [ "my_host" ];
hmHosts = [ "my_other_host" ];

nixosConfigurations = /* some method like genAttrs */ soxin.nixosConfiguration {
  /* ... */
  inherit hmModules;
}:

homeManagerConfigurations = /* some method like genAttrs */ soxin.homeManagerConfiguration {
  /* ... */
  inherit hmModules;
};
```

Comment (kalbasit) [reply lib/home-manager-configuration.nix:7]: sure, I'm fine with the name, let me change it now.

File: lib/home-manager-configuration.nix (hunk 2 above)
Comment (kalbasit) [reply lib/home-manager-configuration.nix:10]: Yes but they are required by home-manager so I decided to expose them on our end so we can document them.

File: lib/home-manager-configuration.nix (hunk 1 above)
Comment (kalbasit) [reply lib/home-manager-configuration.nix]: done

**Review by rissson — APPROVED** [review]

**Pull request merged**